### ISSUE TITLE
Stringifying boolean and defaulting undefined on aria-busy value on Button Component

### ISSUE THREAD

**Issue by tareqlol** [sumup-oss/circuit-ui#1474]
## Component to amend

Button

## Visual

only html changes, nothing will affect the visual.

## Context
the aim of this fix is to solve two issues:
- Providing `undefined` to `isLoading` will skip adding `aria-busy` property.
- ~Providing a `boolean` to `isLoading` will result `aria-busy={false}` instead of `aria-busy="false"` in the compiled html.~ this is unneeded since this [comment](https://github.com/sumup-oss/circuit-ui/issues/1474#issuecomment-1065182398)

**Comment by robinmetral** [comment]
> Providing undefined to isLoading will skip adding aria-busy property.

I think it should actually pass `aria-busy="false"`, so we probably want to make [this](https://github.com/sumup-oss/circuit-ui/blob/5a8e0fc7dd5fb161857a0651012c7aa136344324/packages/circuit-ui/components/Button/Button.tsx#L377) a stringified `Boolean` (just like [this](https://github.com/sumup-oss/circuit-ui/blob/5a8e0fc7dd5fb161857a0651012c7aa136344324/packages/circuit-ui/components/Button/Button.tsx#L384))

**Comment by tareqlol** [comment]
I see, I have a suggestion of making value to be explicit to string like this: 
```
        {...(loadingLabel && {
          'aria-live': 'polite',
          'aria-busy': isLoading ? 'true' : 'false',
        })}
```

**Comment by robinmetral** [comment]
This would work but I'd suggest changing the condition for adding these attributes to the element (currently a truthy check for `loadingLabel`): both aria attributes should only be added if the button can load, i.e. if it has `isLoading = true | false`.

So maybe we can replace your suggestion by:

```diff
-        {...(loadingLabel && {
+        {...(loadingLabel && typeof isLoading === boolean && {
          'aria-live': 'polite',
-          'aria-busy': isLoading ? 'true' : 'false',
+          'aria-busy': isLoading.toString(), // by this point we know `isLoading` is a boolean
        })}
```

Hope I'm not missing something but I think this would work 🙂

**Comment by tareqlol** [comment]
true, I will amend that. thanks!

**Comment by tareqlol** [comment]
@robinmetral I have tried the `.toString()` first before doing my suggestion, it will cause typescript to complain that the you can not assign `aira-busy` to `string | boolean | undefined ` to `Booleanish | undefined` so by explicitly assigning it to  `isLoading ? 'true' : 'false'` solves it with no complain.
![image](https://user-images.githubusercontent.com/61180525/157886998-9286eb17-c75f-4e42-92f9-51b1e1060c95.png)
what do you think, should I go with my change instead, or do you have an idea how to workaround this 🤔 ?

**Comment by robinmetral** [comment]
`isLoading ? 'true' : 'false'` should be fine as well 🙂

**Comment by tareqlol** [comment]
updated
